### PR TITLE
Don't update caches replication stream if tokens haven't advanced

### PR DESCRIPTION
**Pull request by erikjohnston**

### CODE REVIEW

File: synapse/replication/resource.py [hunk 1]
```diff
@@ -390,7 +390,7 @@ def caches(self, writer, current_token, limit, request_streams):
 
         caches = request_streams.get("caches")
 
-        if caches is not None:
+        if caches is not None and current_position != caches:
```
Comment (NegativeMjark) [inline synapse/replication/resource.py]: Note that the other streams use a short-circuit inside the storage method to do this.

**Comment by NegativeMjark** [comment]
Otherwise LGTM

**Comment by NegativeMjark** [comment]
LGTM

**Comment by erikjohnston** [comment]
@matrixbot ok to test

**Pull request merged**